test/e2e/specs/import_gitops_mgmtv3.go

```diff
@@ -229,6 +230,18 @@ func CreateMgmtV3UsingGitOpsSpec(ctx context.Context, inputGetter func() CreateM
 			})
 		}
 
+		if input.TopologyNamespace != "" {
+			Expect(turtlesframework.CreateNamespace(ctx, input.BootstrapClusterProxy, input.TopologyNamespace)).To(Succeed())
+		}
+
+		for _, additionalRepo := range input.AdditionalFleetGitRepos {
+			repoInput := additionalRepo
+			if repoInput.TargetNamespace == "" && input.TopologyNamespace != "" {
+				repoInput.TargetNamespace = input.TopologyNamespace
+			}
+			turtlesframework.FleetCreateGitRepo(ctx, repoInput)
```

Comment:
```suggestion
			repoInput := additionalRepo
			turtlesframework.FleetCreateGitRepo(ctx, repoInput)
```

We should pass these configurations along unaltered.

`TargetNamespace` is the namespace for the `GitRepo` creation. It is either `fleet-local` to automatically apply on local cluster, or some custom generated one - to apply content on the matching “downstream” cluster.

- For the `HelmApp` content it is sufficient to use default `fleet-local` namespace, and specify `TargetNamespace` to the generated one for the test scenario, so that `HelmApp` is created in relation to the Cluster.
- For the `GitRepo` there is a need to specify selectors and it needs to be applied *in* the `TopologyNamespace`. This way it will be evaluated by `cluster` matching, and we don’t need to add `TargetNamespace` there, as there is no reason to have generated namespace on the downstream cluster (it may also not exist and needs to be created).

I’m just feeling that additional defaulting here makes it more complicated to understand what happens.

Reply:
Please checkout https://github.com/rancher/turtles/pull/1203
I think it's also relevant. 

I wonder if we should have one namespace (one per provider) where to put all provider specific classes.
And one additional namespace where to apply all applications.

It does make sense to me to pre-install all providers, classes, and applications before the tests run.
May have the consequence of blocking all tests if one Class does not apply, but we can ignore these errors too.